src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
```

Comment:
its better to iterate array that we get only once, your example do this couple times. we can set a Map of all mothers `const mothersMap = new Map(
    people.filter(p => p.sex === 'f' && p.mother).map(p => [p.name, true])
  );
const mothersMap = new Map(
    people.filter(peson => person.sex === 'f' && peerson.mother).map(person => [person.name, true])
  );
`
then we can check in filter if motherrMap has a name`mothersMap.get(p.mother)`